src/main/java/build/buildfarm/common/io/Directories.java

```diff
@@ -58,6 +59,7 @@ private Directories() {}
   }
 
   public static void remove(Path directory) throws IOException {
+    enableAllWriteAccess(directory);
```

Comment:
I think this as a separate step is unnecessary - having the below walker in a preVisitDirectory set the perms should be the only required activity, it should be confirmed whether java cares about the w bit to delete a file (pretty sure it doesn't).

Reply:
previsit is much better. fixed.